### DISCUSSION TITLE
ConstantAppliedField not using the lower and upper bounds

### DISCUSSION THREAD

**Discussion by Aquios7** [BLAST-WarpX/warpx#4201] [Q&A] [answered]
Hello,
I've been trying to set a grid so that half of the space is affected by a magnetic field using pywarpx with PICMI.

Here is the section of code that I used:
```py
bfield = picmi.ConstantAppliedField(Bx=0.1,
                                    lower_bound=[-0.5, -1, +0],
                                    upper_bound=[+0.5, +3, +3])
```

The grid used is here:
```py
grid = picmi.Cartesian3DGrid(number_of_cells = [nx, ny, nz],
                             lower_bound = [-0.5, -1, -2],
                             upper_bound = [+0.5, +3, +3],
                             lower_boundary_conditions = ['open', 'open', 'open'],
                             upper_boundary_conditions = ['open', 'open', 'open'],
                             lower_boundary_conditions_particles = ['absorbing', 'absorbing', 'absorbing'],
                             upper_boundary_conditions_particles = ['absorbing', 'absorbing', 'absorbing'],
                             warpx_max_grid_size=16)
```

I thought this would give me a b-field in all areas of the grid where z>0, but when I run the code all particles are affected in all areas of the grid.

I don't know if I'm using the code snippet incorrectly or if I need to set a boundary for the field somewhere else.

Any help would be appreciated.
Thank You

**Comment by Aquios7** [comment, marked as answer]
Looking into the source code on the WarpX readthedocs, it looks like the lower and upper bounds are not used by warpx when running PICMI in python. I have changed to an analytic field to get around this issue.

**Reply by ax3l** [reply]
Hi @Aquios7, we agree that is a bug.

Let's transfer this to an issue: #4245
